services/rsources/rsources.go

```diff
@@ -34,15 +34,60 @@ type Stats struct {
 	Failed uint `json:"failed"`
 }
 
-func (r Stats) completed() bool {
+func (r *Stats) completed() bool {
 	return r.In == r.Out+r.Failed
 }
 
+func (r *Stats) corrupted() bool {
+	return r.In < r.Out+r.Failed
+}
+
+func (r *Stats) fixCorrupted() {
+	if r.corrupted() {
+		r.In = r.Out + r.Failed
```

Comment:
What is the reason behind this corruption? Why are we programatically fixing this corruption?

Reply:
This doesn't happen regularly, however there are rare occasions where bugs cause this kind of corruption ([example](https://rudderlabs.slack.com/archives/C02B0A38QQG/p1702224101518269)).

The idea here is to "fix" this corruption virtually for the consuming service to resume without being alerted and fire an alert on our side so that we can further investigate. The alert would be setup on the warning log